Prototype for the Visualize page

DingoEatingFuzz opened the pull request:
This is still a bit rough around the edges, but it represents a realistic living example of what the visualize page can be. 

It is currently using mock data, but the mocks are modeled after the GeoJSON model described in #744 

<img width="1440" alt="Screen Shot 2019-09-09 at 1 18 28 AM" src="https://user-images.githubusercontent.com/174740/64516178-e466fb80-d2a2-11e9-8558-641d27639824.png">

DingoEatingFuzz committed:
Setup the Visualize page

DingoEatingFuzz committed:
Layout the visualize page

DingoEatingFuzz committed:
Add mock contribution data

DingoEatingFuzz committed:
Implement the three data visualizations, including a map

DingoEatingFuzz committed:
Prettier etc.

danmelton committed:
adding format geoJson to getContributions

danmelton commented on api/models/entity/Contribution.ts:
```diff
@@ -531,12 +531,43 @@ export const contributionSummaryFields = <const>[
+        city: string;
+        state: string;
+        zip: string;
+        amount: number;
```
add matchAmount, oaeType

danmelton commented on api/models/entity/Contribution.ts:
```diff
@@ -531,12 +531,43 @@ export const contributionSummaryFields = <const>[
+        date: string;
+        campaign: {
+            name: string;
+            id: string
```
add office

danmelton commented on api/models/entity/Contribution.ts:
```diff
@@ -609,10 +643,44 @@ export async function getContributionsByGovernmentIdAsync(
+export function convertToGeoJson(contributions: IContributionSummaryResults): IContributionsGeoJson {
+    const data  = contributions
+        .data
+        .map((contribution: Pick<Contribution, typeof contributionSummaryFields[number]>): IContributionGeoJson => {
```
add filter for archive/draft

danmelton commented on api/models/entity/Contribution.ts:
```diff
@@ -609,10 +643,44 @@ export async function getContributionsByGovernmentIdAsync(
+            return {
+                type: 'Feature',
+                properties: {
+                    city: contribution.city,
```
add matchAmount and oaeType

danmelton committed:
fixing rebase conflict

danmelton committed:
adds api geojson endpoint closes gh-744

danmelton committed:
adds redux connector for getting public data gh-974

danmelton commented:
This feature needs all the gis functinoality of the app, and contributions that are processed or submitted.

SO. First, let's setup the api.

You'll need a .env file (copy the /.env.example to /.env, and add the GIS key - ask dan for it)

docker-compose up api (let this run)
It'll seed the db, and populate the redis queue for our jobs service

build and run the docker jobs
docker-compose build jobs
docker-compose up jobs (let it run - it will do our geocoding)

cd app
yarn install && yarn start

navigate to frontend UI
login as govadmin@openelectionsportland.org/password
navigate to contributions, view any contribution, update the status in the upper right hand corner to submitted

you will have data at :3000/contributionsgeo

the HomePage component has an example of the redux action to grab publicData.

DingoEatingFuzz committed:
TEMP: A quick update all action to mark all contributions as submitted

DingoEatingFuzz committed:
Beginning of dataviz selectors

Including public data, filter values, filter selections, and filtered
public data.

DingoEatingFuzz committed:
Add start and end dates to filters

DingoEatingFuzz committed:
Outline the various views of the filtered data required for each viz type

DingoEatingFuzz committed:
Test coverage and the summary selector

DingoEatingFuzz committed:
Aggregate views for public data visualizations

DingoEatingFuzz committed:
Campaigns table selector

jaronheard commented:
@danmelton I'm working on updating some of the seed data so that it has a mix of campaigns and contribution type/subtype so that we have some better sample data for the visualizations, and was wondering if setting some contribution statuses to submitted in the seed file would worry you for any reason.

I don't think it's absolutely necessary, so let me know if you think it would cause issues.

danmelton commented:
> @danmelton I'm working on updating some of the seed data so that it has a mix of campaigns and contribution type/subtype so that we have some better sample data for the visualizations, and was wondering if setting some contribution statuses to submitted in the seed file would worry you for any reason.
> 
> I don't think it's absolutely necessary, so let me know if you think it would cause issues.

@jaronheard Yep, go for it.

DingoEatingFuzz committed:
Actions and reducers for setting filters

jaronheard committed:
Updated seeds to include multiple campaigns and contribution types

* Added another campaign and gave permissions to that campaign
* Randomized contributions between those campaigns
* Randomized contributor type, contribution type and sub-type

jaronheard commented:
@danmelton re-opening this since we're still working off this branch, though if you've got another suggested approach let me know.

jaronheard committed:
Refactor campaign permission logic in seeds.ts for extensibility

jaronheard committed:
update component-library to latest

DingoEatingFuzz committed:
Add campaigns table to the home page

DingoEatingFuzz committed:
Contributions map

jaronheard committed:
add basic contribution pie charts

DingoEatingFuzz committed:
Office and campaign filters

jaronheard committed:
fix prettier config

DingoEatingFuzz committed:
Style the filters

DingoEatingFuzz committed:
Fix the comically long table

jaronheard committed:
Refactor contributor data transform to selectors

jaronheard committed:
Set table to respect paging property value

jaronheard committed:
Add summary table and add count pie charts

jaronheard committed:
update public data to avoid direct victory labels

jaronheard committed:
update home visualization layout

jaronheard committed:
[potentially temp] add value count toggle

DingoEatingFuzz committed:
Swap out the jpg logo with an svg one

jaronheard committed:
implement by location visualization and comment out radio button

DingoEatingFuzz committed:
Sentence structure for filters

jaronheard committed:
update tests and add contributor bar

DingoEatingFuzz committed:
Date range filter for the public viz page

DingoEatingFuzz committed:
Add date range filter to the public viz page

DingoEatingFuzz committed:
Move the amount/count toggle up to the sentence

jaronheard committed:
Revert "TEMP: A quick update all action to mark all contributions as submitted"

This reverts commit 1ea10478176a57c7328fe01bc6790ec9a047afe2.

jaronheard committed:
Revert "Refactor campaign permission logic in seeds.ts for extensibility"

This reverts commit 09d589a9556a62060801f529bd3822065730f670.

jaronheard committed:
Revert "Updated seeds to include multiple campaigns and contribution types"

This reverts commit 6381c926499d42bf967a36479cca6d02d3aff79a.

jaronheard committed:
Update selector tests

jaronheard committed:
Bump component-library version

jaronheard committed:
Update Home test snapshot

jaronheard committed:
Merge remote-tracking branch 'origin/develop' into prototype-visualizations

jaronheard committed:
check for null options
